src/ui/Editor.cpp

```diff
@@ -1854,8 +1855,42 @@ Editor::_SetFoldMargin(bool enabled)
 void
 Editor::SetProblems()
 {
-	BAutolock lock(fProblemsLock);
+	// make absolutely sure we're locked
+	if (!Window()->IsLocked()) {
+		debugger("The looper must be locked !");
```

Comment:
Do we really want to summon the debugger here? Wouldn't be better to just raise an alert and bail out?

Reply:
This is for the developer (us) and not for the user : just like calling GetMouse with the looper unlocked.
You can see this like an assert ().
If we just bail out it would be harder to track down wrong use of this api